libpromises/rlist.c

```diff
@@ -373,13 +373,14 @@ Rval RvalNewRewriter(const void *item, RvalType type, JsonElement *map)
 
             Buffer *format = BufferNew();
             StringCopy(item, buffer_from, max_size);
+            buffer_to[0] = '\0';
 
             for (int iteration = 0; iteration < 10; iteration++)
             {
                 bool replacement_made = false;
                 int var_start = -1;
                 char closing_brace = 0;
-                for (int c = 0; c < buffer_from[c]; c++)
```

Comment:
If I'm not missing something this does not make sense and risks stopping too early (if the input is long and its size reaches values near 0-9/a-z ascii values).